pkg/controller/config-controller.go

```diff
@@ -0,0 +1,396 @@
+		return
+	}
+	for _, conf := range configs {
+		if conf.Name == c.configName {
+			config = conf
+		}
+	}
+
+	if config == nil {
+		runtime.HandleError(errors.Errorf("error getting CDI config"))
+		return
+	}
+
+	if ing, ok := obj.(*extensionsv1beta1.Ingress); ok {
+		for _, rule := range ing.Spec.Rules {
+			for _, path := range rule.HTTP.Paths {
+				if path.Backend.ServiceName == c.uploadProxyServiceName {
+					c.enqueueCDIConfig(config)
+					return
+				}
+			}
+		}
+	}
+	if route, ok := obj.(*routev1.Route); ok {
+		if route.Spec.To.Name == c.uploadProxyServiceName {
+			c.enqueueCDIConfig(config)
+			return
+		}
+	}
+	if conf, ok := obj.(*cdiv1.CDIConfig); ok {
+		if conf.Name == config.Name {
+			c.enqueueCDIConfig(conf)
+			return
+		}
+	}
+	return
+}
+
+func (c *ConfigController) runWorker() {
+	for c.processNextWorkItem() {
+	}
+}
+
+func (c *ConfigController) processNextWorkItem() bool {
+	obj, shutdown := c.queue.Get()
+
+	if shutdown {
+		return false
+	}
+
+	err := func(obj interface{}) error {
+		defer c.queue.Done(obj)
+
+		var key string
+		var ok bool
+
+		if key, ok = obj.(string); !ok {
+			c.queue.Forget(obj)
+			runtime.HandleError(errors.Errorf("expected string in workqueue but got %#v", obj))
+			return nil
+		}
+
+		if err := c.syncHandler(key); err != nil {
+			return errors.Errorf("error syncing '%s': %s", key, err.Error())
+		}
+
+		c.queue.Forget(obj)
+		glog.Infof("Successfully synced '%s'", key)
+		return nil
+
+	}(obj)
+
+	if err != nil {
+		runtime.HandleError(err)
+		return true
+	}
+
+	return true
+}
+
+func (c *ConfigController) syncHandler(key string) error {
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		runtime.HandleError(errors.Errorf("invalid resource key: %s", key))
+		return nil
+	}
+
+	selector, err := metav1.LabelSelectorAsSelector(&metav1.LabelSelector{})
+	if err != nil {
+		return err
+	}
+
+	var config *cdiv1.CDIConfig
+	configs, err := c.configLister.CDIConfigs(namespace).List(selector)
+	if err != nil {
+		return err
+	}
+	for _, conf := range configs {
+		if conf.Name == name {
+			config = conf
+		}
+	}
+
+	if config == nil {
+		runtime.HandleError(errors.Errorf("CDIConfig '%s' in work queue no longer exists", key))
+		return nil
+	}
+
+	var url string
+
+	routes, err := c.routeLister.List(selector)
+	if err != nil {
+		return err
+	}
+	for _, route := range routes {
+		url = getURLFromRoute(route, c.uploadProxyServiceName)
+		if url != "" {
+			break
+		}
+	}
+	if url == "" {
+		ingresses, err := c.ingressLister.List(selector)
+		if err != nil {
+			return err
+		}
+		for _, ing := range ingresses {
+			url = getURLFromIngress(ing, c.uploadProxyServiceName)
+			if url != "" {
+				break
+			}
+		}
+	}
+	//if user overrode the upload proxy url, ignore all routes and ingresses
+	if config.Spec.UploadProxyURLOverride != nil && config.Spec.UploadProxyURLOverride != config.Status.UploadProxyURL {
```

Comment:
Not sure I understand this part of `if` condition `config.Spec.UploadProxyURLOverride != config.Status.UploadProxyURL`

Is it your intention to compare to pointer values?

Also, wouldn't you always want to assign `url` to `*config.Spec.UploadProxyURLOverride` if it is set?

Reply:
correct, thanks!